crates/shim/src/mount.rs

```diff
@@ -723,4 +723,47 @@ mod tests {
             assert_eq!(options, expected_options);
         }
     }
+
+    #[test]
+    fn test_mount_rootfs() {
+        let fs_type = Some("overlay");
+        let source = Some("overlay");
+        let workdir =
+            "/var/lib/containerd-test/io.containerd.snapshotter.v1.overlayfs/snapshots/0/work";
+        std::fs::create_dir_all(workdir).unwrap();
+        let upperdir =
+            "/var/lib/containerd-test/io.containerd.snapshotter.v1.overlayfs/snapshots/0/fs";
+        std::fs::create_dir_all(upperdir).unwrap();
+        let mut dirs: Vec<String> = Vec::with_capacity(125);
+        for i in 1..=125 {
+            let mut dir =
+                "/var/lib/containerd-test/io.containerd.snapshotter.v1.overlayfs/snapshots/"
+                    .to_string()
+                    + i.to_string().as_str()
+                    + "/fs";
+            std::fs::create_dir_all(&dir).unwrap();
+            dirs.push(dir);
+        }
+        let options = vec![
+            "index=off".to_string(),
+            "workdir=".to_string() + workdir,
+            "upperdir=".to_string() + upperdir,
+            "lowerdir=".to_string() + dirs.join(":").as_str(),
+        ];
+
+        let target = "/run/containerd/io.containerd.runtime.v2.task/k8s.io/mount-test/rootfs";
+        std::fs::create_dir_all(target).unwrap();
+        let current_dir = env::current_dir().unwrap();
+        std::thread::spawn(move || mount_rootfs(fs_type, source, &options, target).unwrap());
```

Comment:
Why must it spawn thread to mount it?

Reply:
because this mount_rootfs is called in a new thread by tokio::spawn_blocking, so I think it's better to test mount_rootfs in a spawn thread.